great_expectations/core/batch_spec.py

```diff
@@ -107,3 +107,21 @@ def query(self):
     @query.setter
     def query(self, query) -> None:
         self["query"] = query
+
+
+class GlueCatalogBatchSpec(BatchSpec, metaclass=ABCMeta):
+    """This is an abstract class and should not be instantiated. It's relevant for testing whether
+    a subclass is allowed
+    """
+
+    @property
+    def database_name(self) -> str:
+        return self.get("database_name", None)
+
+    @property
+    def table_name(self) -> str:
+        return self.get("table_name", None)
+
+    @property
+    def query_condition(self) -> str:
+        return self.get("query_condition", None)
```

Comment:
@lccasagrande Quick question: What is the definition of "query_condition" -- is it a must have for all descendants?  Do we need a docstring here?  By the way, all public methods are required to have a docstring and a return type.  Thanks.

Reply:
@lccasagrande Highlighting the following -- our code checkers flag any occurrences of a docstring and the return type (including `None`) not present for public methods (so please check -- this might eliminate many test failures).  Thank you.

Reply:
I removed this as it is not required anymore. I am using AWS Glue Data Catalog to get batch identifiers from the table partitions. Therefore, it can be filtered by the partitions or using the already available methods allowed in batch_spec_passthrough.